pkg/cli/admin/prune/renderedmachineconfigs/renderedmachineconfigs.go

```diff
@@ -0,0 +1,443 @@
+	inuseConfigs, err := o.getInUseConfigs(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Sort the machine configs based on the flags provided
+	if o.PoolFilter == "" {
+		if o.Count > 0 {
+			o.sortMachineConfigsByTime(machineConfigs)
+		} else {
+			o.sortMachineConfigs(machineConfigs)
+		}
+	} else {
+		o.sortMachineConfigsByTime(machineConfigs)
+	}
+
+	// Delete the oldest rendered configs based on the count
+	countToDelete := o.Count
+	if countToDelete < 0 {
+		return errors.New("count cannot be negative")
+	}
+	if countToDelete == 0 || countToDelete > len(machineConfigs) {
+		countToDelete = len(machineConfigs)
+	}
+
+	if !o.Confirm {
+		fmt.Fprintln(o.IOStreams.ErrOut, "Dry run enabled - no modifications will be made. Add --confirm to remove rendered machine configs.")
+	}
+
+	for i := 0; i < countToDelete; i++ {
+		if !inuseConfigs.Has(machineConfigs[i].Name) {
+			if err := o.deleteRenderedConfig(ctx, machineConfigs[i].Name, o.Confirm); err != nil {
+				fmt.Fprintf(o.IOStreams.ErrOut, "Error deleting config %s: %v \n", machineConfigs[i].Name, err)
+			}
+		} else {
+			fmt.Fprintf(o.IOStreams.ErrOut, "\nSkipping deletion of %s as it's currently in use\n", machineConfigs[i].Name)
+		}
+	}
+
+	return nil
+}
+
+func (o *pruneMCOptions) RunList(ctx context.Context, args []string) error {
+	// Check the --in-use flag
+	if o.InUse {
+		// List machine config pools for in-use rendered configs
+		return o.listInUseRenderedConfigs(ctx)
+	}
+	// List machine config pools for in-use rendered configs
+	machineConfigs, err := o.listRenderedConfigs(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Call the sortAndPrintMachineConfigs function to sort and print the MachineConfig structs
+	o.sortMachineConfigs(machineConfigs)
+	o.printMachineConfigs(machineConfigs)
+
+	return nil
+}
+
+// extractPoolName retrieves the pool name from the owner references of the machine config
+func extractPoolName(meta *metav1.ObjectMeta) string {
+	ownerReferences := meta.GetOwnerReferences()
+	for _, ownerRef := range ownerReferences {
+		return ownerRef.Name
+	}
+	return "unknown" // fallback value if no owner reference is found
+}
+
+// MachineConfig is a struct representing a machine config
+type MachineConfig struct {
+	Name      string
+	CreatedAt time.Time
+	Pool      string
+	IsCurrent bool
+}
+
+func (o *pruneMCOptions) listInUseRenderedConfigs(ctx context.Context) error {
+	clientConfig, err := o.RESTClientGetter.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	machineConfigClient, err := mcfgclientset.NewForConfig(clientConfig)
+	if err != nil {
+		return err
+	}
+
+	poolList, err := machineConfigClient.MachineconfigurationV1().MachineConfigPools().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("getting machine config pools failed: %w", err)
+	}
+
+	found := false
+
+	for _, pool := range poolList.Items {
+		// Get the rendered config name from the status section
+		renderedConfigName := pool.Status.Configuration.Name
```

Comment:
I think for inuse rendered MC, we want to list all MachieConfigs that are still in use by any node as well as ? Or this is more about showing what is in the MCP status? 
Also, I think we would want to look at pool.Spec.Configuration.Name instead of pool.Spec.Configuration.Name . Spec contains what the node in the pool should be updated to.

Reply:
Thanks David for already updating the offline conversation.